packages/ember-metal/lib/streams/stream.js

```diff
@@ -40,6 +41,8 @@ BasicStream.prototype = {
     this.dependencyHead = null;
     this.dependencyTail = null;
     this.observedProxy = null;
+    this.__ember_meta__ = null;
+    this[GUID_KEY] = null;
```

Comment:
we should likely follow:

1. undefined for empty slot
2. null for no data in a given slot